src/integration/test_suites/smoke_test_dma_aes_gcm_short_dword/smoke_test_dma_aes_gcm_short_dword.c

```diff
@@ -0,0 +1,275 @@
+    int len = strlen(hex_str);
+    int num_dwords;
+    int num_chars;
+
+    VPRINTF(HIGH, "String length is %d.\n", len);
+    const uint32_t index[] = {1, 0, 3, 2, 5, 4, 7, 6};
+    if (len % 2 != 0) {
+        VPRINTF(HIGH, "Error: Hex string length must be a multiple of 2.\n");
+        return;
+    }
+    num_dwords = (len / 8);
+    *array_size = (len / 2);
+    for (int i = 0; i <= num_dwords; i++) {
+        uint32_t value = 0x00000000;
+        num_chars = (i == num_dwords) ? len % 8 : 8;
+        for (int j = 0; j < num_chars; j++) {
+            char c = hex_str[i * 8 + j];
+            uint32_t digit;
+
+            if (c >= '0' && c <= '9') {
+                digit = c - '0';
+            } else if (c >= 'a' && c <= 'f') {
+                digit = c - 'a' + 10;
+            } else if (c >= 'A' && c <= 'F') {
+                digit = c - 'A' + 10;
+            } else {
+                VPRINTF(HIGH, "Error: Invalid hex character: %c\n", c);
+                return;
+            }
+            value |= digit << (4 * index[j]);
+        }
+        if (num_chars != 0) {
+            array[i] = value;
+        }
+    }
+
+    // Apply endianness if needed
+    if (endian_mode == AES_BIG_ENDIAN) {
+        for (int i = 0; i < *array_size; i++) {
+            array[i] =  (((array[i] & 0xFF000000) >> 24) |
+                        ((array[i] & 0x00FF0000) >> 8)  |
+                        ((array[i] & 0x0000FF00) << 8)  |
+                        ((array[i] & 0x000000FF) << 24));
+        }
+    }
+}
+
+void run_aes_test(test_config_t test_config) {
+    aes_flow_t aes_input;
+    aes_op_e op = test_config.operation;
+    aes_mode_e mode = test_config.mode;
+    aes_key_len_e key_len;
+
+    VPRINTF(LOW, "\n----------------------------------\n%s\n----------------------------------\n", test_config.test_name);
+
+    const char key_str3[] = "feffe9928665731c6d6a8f9467308308feffe9928665731c6d6a8f9467308308";
+    const char ecb_key_str[] = "b6a8d5636f5c6a7224f9977dcf7ee6c7fb6d0c48cbdee9737a959796489bddbc";
+    uint32_t key[8];
+    uint32_t key_size;
+                    
+    const char iv_str3[] = "cafebabefacedbaddecaf888";
+    uint32_t iv[4]; 
+    uint32_t iv_length;
+
+    
+    const char plaintext_str3[] = "d9313225f88406e5a55909c5aff5269a86a7a9531534f7da2e4c303d8a318a721c3c0c95956809532fcf0e2449a6b525b16aedf5aa0de657ba637b39";
+    uint32_t plaintext[32]; //arbitrary length here
+    uint32_t plaintext_length;
+
+    const char ciphertext_str3[] = "522dc1f099567d07f47f37a32a84427d643a8cdcbfe5c0c97598a2bd2555d1aa8cb08e48590dbb3da7b08b1056828838c5f61e6393ba7a0abcc9f662";
+    uint32_t ciphertext[32]; //arbitrary length here
+    uint32_t ciphertext_length;
+
+    const char aad_str3[] = "feedfacedeadbeeffeedfacedeadbeefabaddad2";
+    uint32_t aad[32]; //arbitrary length here
+    uint32_t aad_length;
+                    
+    const char tag_str3[] = "76fc6ece0f4e1768cddf8853bb2d551b";
```

Comment:
Are you able to have a fixed tag given the random size of your data? I thought this was a problem?